### PR TITLE
Bug 1115508 - [Flame][Built-in&System Keyboard]The utility bubble always shows even at lock screen.

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1115508

### CODE REVIEW

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=fd53d8b7dd1f)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=83ef3babde00)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f9246dbbb509)

File: apps/system/js/app_window_manager.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* global SettingsListener, homescreenWindowManager, inputWindowManager,
           layoutManager, Service, NfcHandler, rocketbar, ShrinkingUI,
-          FtuLauncher */
+          FtuLauncher, textSelectionDialog */
```
Comment (alivedise) [inline apps/system/js/app_window_manager.js]: Remove this change

File: apps/system/js/app_window_manager.js [hunk 2]
```diff
@@ -44,10 +44,19 @@
...
       if (active) {
         this.focus();
+      } else {
+        this.blur();
       }
       this._activeApp.setVisibleForScreenReader(active);
     },
 
+    blur: function() {
+      if (textSelectionDialog && textSelectionDialog.isActive()) {
```
Comment (alivedise) [inline apps/system/js/app_window_manager.js]: textSelectionDialog should do this on its own. AppWindowManager is overloaded.

File: apps/system/js/text_selection_dialog.js [hunk 3]
```diff
@@ -62,11 +62,16 @@
...
+  TextSelectionDialog.prototype.isActive = function tsd_isActive() {
+    return this._enabled && this._transitionState !== 'closed';
+  };
+
   TextSelectionDialog.prototype.start = function tsd_start() {
     if (this._enabled) {
       return;
     }
     this._enabled = true;
+    window.addEventListener('screenchange', this);
```
Comment (alivedise) [inline apps/system/js/text_selection_dialog.js]: Where is handler of this event?

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2c08f580c3b1)

**Pull request merged**